odoo/addons/base/models/ir_property.py

```diff
@@ -113,9 +113,9 @@ def write(self, values):
         if self._ids:
             self.env.cr.execute(
                 'SELECT EXISTS (SELECT 1 FROM ir_property WHERE id in %s AND res_id IS NULL)', [self._ids])
-            default_set = self.env.cr.rowcount == 1 or any(
-                v.get('res_id') is False
-                for v in values
+            default_set = self.env.cr.rowcount == 1 and self.env.cr.fetchone()[0] or any(
+                v.res_id is False
+                for v in self
```

Comment:
Not sure we should check `self` here, as the current data in DB are already checked by the SQL query, and we need to check the "we're writing a res_id=False on any record" from the comment above:
```suggestion
            default_set = self.env.cr.rowcount == 1 and self.env.cr.fetchone()[0] or values.get("res_id") is False
```
What do you think?